rust/operator-binary/src/controller.rs

```diff
@@ -1115,3 +1253,24 @@ pub fn build_recommended_labels<'a, T>(
         role_group,
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_get_most_expressive_log_level() {
+        let log_level = get_most_expressive_log_level(&[]);
+        assert!(log_level.is_none());
+
+        let log_level = get_most_expressive_log_level(&[LogLevel::ERROR]);
+        assert_eq!(log_level.unwrap(), LogLevel::ERROR);
+
+        let log_level = get_most_expressive_log_level(&[LogLevel::INFO, LogLevel::ERROR]);
+        assert_eq!(log_level.unwrap(), LogLevel::INFO);
+
+        let log_level =
+            get_most_expressive_log_level(&[LogLevel::INFO, LogLevel::INFO, LogLevel::NONE]);
+        assert_eq!(log_level.unwrap(), LogLevel::INFO);
+    }
+}
```

Comment:
note: These tests can probably be removed, or moved to `stackable-operator` if you want to ensure the ordering is consistent.